File: python3/RansomNote.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        memo = {}
        
        for char in magazine:
            if char in memo:
                memo[char] += 1
            else:
                memo[char] = 1
        
        for char in ransomNote:
            if char not in memo:
                return False
            else:
                if memo[char] <= 0:
                    return False
                memo[char] -= 1
        
        return True

File: c++/RansomNote.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int map[26] = {0};
        
        for (char c : magazine) {
            map[c-'a'] += 1;
        }
        
        for (char c : ransomNote) {
            if (map[c - 'a'] <= 0) {
                return false;
            }
            map[c - 'a'] -= 1;
        }
        
        return true;
    }
};
